src/commands/dev/mod.rs

```diff
@@ -43,170 +20,24 @@ pub fn dev(
-                    let resp = Response::from_parts(parts, body);
-
-                    println!(
-                        "[{}] {} {}{} {:?} {}",
-                        now.format("%Y-%m-%d %H:%M:%S"),
-                        req_method,
-                        server_config.host,
-                        path,
-                        version,
-                        resp.status()
-                    );
-                    Ok::<_, failure::Error>(resp)
-                }
-            }))
-        }
-    });
-
-    let server = Server::bind(&listening_address.address).serve(make_service);
-    println!("{} Listening on http://{}", emoji::EAR, listening_address);
-    if let Err(e) = server.await {
-        eprintln!("server error: {}", e);
-    }
-    Ok(())
-}
-
-fn get_preview_url(path_string: &str) -> Result<Uri, InvalidUri> {
-    format!("https://{}{}", PREVIEW_HOST, path_string).parse()
-}
-
-fn get_path_as_str(uri: &Uri) -> String {
-    uri.path_and_query()
-        .map(|x| x.as_str())
-        .unwrap_or("")
-        .to_string()
-}
-
-fn preview_request(
-    req: Request<Body>,
-    client: HyperClient<HttpsConnector<HttpConnector>>,
-    preview_id: String,
-) -> ResponseFuture {
-    let (mut parts, body) = req.into_parts();
-
-    let path = get_path_as_str(&parts.uri);
-    let preview_id = &preview_id;
-
-    structure_request(&mut parts);
-
-    parts.headers.insert(
-        HeaderName::from_static("host"),
-        HeaderValue::from_static(PREVIEW_HOST),
-    );
-
-    parts.headers.insert(
-        HeaderName::from_static("cf-ew-preview"),
-        HeaderValue::from_str(preview_id).expect("Could not create header for preview id"),
-    );
-
-    parts.uri = get_preview_url(&path).expect("Could not get preview url");
-
-    let req = Request::from_parts(parts, body);
-
-    client.request(req)
-}
+    // before serving requests we must first build the Worker
+    commands::build(&target)?;
 
-fn get_session_id() -> Result<String, failure::Error> {
-    Ok(Uuid::new_v4().to_simple().to_string())
+    // eventually we will have two modes - edge and gcs
+    // edge for authenticated users and gcs for unauthenticated
+    // for now, always route to gcs
+    gcs::dev(target, user, server_config, verbose)
 }
 
-pub fn get_preview_id(
-    mut target: Target,
-    user: Option<GlobalUser>,
-    server_config: &ServerConfig,
-    session_id: &str,
-    verbose: bool,
-) -> Result<String, failure::Error> {
-    let sites_preview = false;
-    let script_id = upload(&mut target, user.as_ref(), sites_preview, verbose).map_err(|_| failure::format_err!("Could not upload your script. Check your internet connection or https://www.cloudflarestatus.com/ for rare incidents impacting the Cloudflare Workers API."))?;
-    Ok(format!(
-        "{}{}{}{}",
-        &script_id,
-        session_id,
-        server_config.host.is_https() as u8,
-        server_config.host
-    ))
+fn print_alpha_warning_message() {
```

Comment:
not sure if it matters as much here, but we may  consider  thinking more closely about  when we print to stdout vs stderr; this is for sure a  thing in tail logs but could use further  consideration across the codebase.